src/missingarray.jl

```diff
@@ -26,3 +26,13 @@ Base.fill!(A::CategoricalArray{>:Missing}, ::Missing) = (fill!(A.refs, 0); A)
 
 in(x::Missing, y::CategoricalArray) = false
 in(x::Missing, y::CategoricalArray{>:Missing}) = !all(v -> v > 0, y.refs)
+
+function Missings.replace(a::CategoricalArray{S, N, R, V, C}, replacement::V) where {S, N, R, V, C}
+    pool = CategoricalPool(a.pool)
+    v = C(get!(pool, replacement), pool)
+    Missings.replace(a, v)
+end
+
+function collect(r::Missings.EachReplaceMissing{CategoricalArray{S, N, R, C, V, U},V}) where {S, N, R, C, V, U}
```

Comment:
Could probably get rid of type parameters and do `<:CategoricalArray` instead.

Reply:
I still need R from the type parameters to work with empty arrays

Reply:
Good catch. Though you can drop the following parameters.

Reply:
I also need V for the second type argument in `Missings.EachReplaceMissing`

Reply:
But you don't have to specify that type since IIRC it's always converted to the element type of the array.

Reply:
You're right, I could drop the second type argument to `EachReplaceMissing`. But it doesn't work for me if I drop any more type parameters.

Example:
```
> function testcollect(r::Missings.EachReplaceMissing{CategoricalArray{S, N, R, C}}) where {S, N, R, C}
    CategoricalArray{C,N}([v.level for v in r], r.replacement.pool)
end
> x = categorical(["a", "b", missing, "a"])
> r = Missings.replace(x, "")
> testcollect(r)
ERROR: MethodError: no method matching testcollect(::Missings.EachReplaceMissing{CategoricalArrays.CategoricalArray{Union{Missings.Missing, String},1,UInt32,String,CategoricalArrays.CategoricalString{UInt32},Missings.Missing},CategoricalArrays.CategoricalString{UInt32}})
Closest candidates are:
  testcollect(::Missings.EachReplaceMissing{CategoricalArrays.CategoricalArray{S,N,R,C,C,U} where U where C,U} where U) where {S, N, R, C}
```

Reply:
You need `<:CategoricalArray{...}`. Else you're asking for a type with only some parameters specified.